app/models/searchgov_crawler.rb

```diff
@@ -68,10 +70,15 @@ def application_extensions
     %w{doc docx pdf xls xlsx ppt pptx} #support powerpoint
   end
 
-  def skiplinks_regex
+  def skip_extensions_regex
     /\.(#{(Fetchable::BLACKLISTED_EXTENSIONS + application_extensions ).join('|')})$/i
   end
 
+  # avoid infinite loops caused by malformed urls
+  def repeating_segments_regex
+    /(([^\/]+\/)+)\1\1/
```

Comment:
@noremmie , this is similar to the regex used by Nutch: https://github.com/apache/nutch/blob/master/conf/regex-urlfilter.txt.template#L35-L36.  I tested this against our nearly 1 million SG urls, and the only false positive was https://www.fmcsa.dot.gov/regulations/title49/any/any/any/list, which appears to have been replaced by https://www.fmcsa.dot.gov/regulations/title49/b/5/3/list. Otherwise, it captures exactly what I want to avoid, which is links like:
https://www.nrcs.usda.gov/wps/portal/nrcs/newsarchives/al/newsroom/releases/newsarchives/www.twitter.com/NRCS_AL/www.twitter.com/NRCS_AL/www.twitter.com/NRCS_AL/www.twitter.com/NRCS_AL/www.twitter.com/NRCS_AL/www.twitter.com/NRCS_AL/www.twitter.com/NRCS_AL/....etc.

Reply:
I like your empirical approach! This sounds good to me, and I'm guessing that if this leads to any problems they will either be minor or immediately noticeable 👍